ricecooker/utils/browser.py

```diff
@@ -1,8 +1,62 @@
-import os, urllib, posixpath, webbrowser
-from http.server import HTTPServer, BaseHTTPRequestHandler, SimpleHTTPRequestHandler
+import os
+import posixpath
+import sys
+import urllib
+import webbrowser
 
+from http.server import HTTPServer, HTTPStatus, SimpleHTTPRequestHandler
 
-def preview_in_browser(directory, filename="index.html", port=8282):
+SERVER = "127.0.0.1"
+PORT = 8282
+
+
+class IFrameServerRequestHandler(SimpleHTTPRequestHandler):
+    def do_OPTIONS(self):
+        self.send_access_control_headers()
+        self.send_response(200, "ok")
+
+    def send_access_control_headers(self):
+        self.send_header('Access-Control-Allow-Origin', '*')
+        self.send_header('Access-Control-Allow-Methods', 'GET, OPTIONS')
+        self.send_header("Access-Control-Allow-Headers", "X-Requested-With")
+
+    def do_GET(self):
+        path = self.translate_path(self.path)
+        print("path = {}".format(path))
+        ctype = self.guess_type(path)
+        try:
+            f = open(path, 'rb')
+        except OSError:
+            self.send_error(HTTPStatus.NOT_FOUND, "File not found")
+            return None
+        try:
+            self.send_response(HTTPStatus.OK)
+            self.send_header("Content-type", ctype)
+            fs = os.fstat(f.fileno())
+            self.send_header("Content-Length", str(fs[6]))
```

Comment:
Isn't there an attribute for the file size on `fs`? Might be clearer.

Reply:
Yeah, I just pushed a change for this, thanks!